lib/ember-qunit/qunit-module.js

```diff
@@ -45,21 +35,24 @@ export function createModule(Constructor, name, description, callbacks) {
   var module = new Constructor(name, description, callbacks);
 
   qunitModule(module.name, {
-    setup: function(assert) {
-      var done = assert.async();
-      return module.setup().then(function() {
+    beforeEach() {
+      return module.setup(...arguments).then(() => {
+        Object.keys(module.context).forEach((key) => {
```

Comment:
Lets extract this into https://github.com/rwjblue/ember-qunit/pull/237 for now. I'd prefer to just use QUnit's test context directly as mentioned in a comment over there.  I definitely do not see this as a blocker either way though.